### PR TITLE
Try to fix Monitor Query tests hangs again

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Integration.Tests/FunctionalTests/AzureMonitorLogExporterLiveTests.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@ public class AzureMonitorLogExporterLiveTests : AzureMonitorTestBase
         public AzureMonitorLogExporterLiveTests(bool isAsync) : base(isAsync) { }
 
         [RecordedTest]
+        [PlaybackOnly("https://github.com/Azure/azure-sdk-for-net/issues/24360")]
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Integration.Tests/FunctionalTests/AzureMonitorLogExporterLiveTests.cs:25]: cc @vishweshbankwar

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientSamples.cs [hunk 2]
```diff
@@ -15,6 +15,7 @@ namespace Azure.Monitor.Query.Tests
     public class LogsQueryClientSamples: SamplesBase<MonitorQueryTestEnvironment>
     {
         [Test]
+        [Explicit]
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientSamples.cs:18]: AzureActivity table is not always available causing tests to fail when running live. We have the functionality covered in live tests.

File: sdk/monitor/Azure.Monitor.Query/tests/MetricsQueryClientLiveTests.cs [hunk 3]
```diff
@@ -258,11 +258,6 @@ public async Task CanListNamespacesMetrics()
                 ns.Name == "Microsoft.OperationalInsights-workspaces" &&
                 ns.Type == "Microsoft.Insights/metricNamespaces" &&
                 ns.FullyQualifiedName == "Microsoft.OperationalInsights/workspaces"));
-
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/tests/MetricsQueryClientLiveTests.cs:261]: https://github.com/Azure/azure-sdk-for-net/issues/24361

File: sdk/monitor/Azure.Monitor.Query/tests/MetricsTestData.cs [hunk 4]
```diff
@@ -100,40 +106,40 @@ private async Task SendData(MetricsSenderClient senderClient)
 
         private async Task<bool> MetricsPropagated(MetricsQueryClient metricQueryClient)
         {
-            var nsExists =  (await metricQueryClient.GetMetricNamespacesAsync(_testEnvironment.MetricsResource).ToEnumerableAsync()).Any(ns => ns.Name == MetricNamespace);
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/tests/MetricsTestData.cs:103]: https://github.com/Azure/azure-sdk-for-net/issues/24361

**Review by vishweshbankwar — APPROVED** [review]

**Review by nisha-bhatia — APPROVED** [review]

**Pull request merged**